js/forgotPassword.js

```diff
@@ -0,0 +1,93 @@
+$(document).ready(() => {
+    $("#reNewPass").hide();
+    $("#resPass").click(() => {
+        let ExistEmail = false;
+        for (let i of usersArray) {
+            if (i.Email == $("#checkEmail").val()) {
+                ExistEmail = true;
+                break;
+            }
+        }
+        if (ExistEmail) {
+            $("#checkEmailDiv").hide();
+            $("#reNewPass").show();
+        }
+        else {
+            $("#ErrorMsg").text("Email doesn't Exist").css("color", "red");
+        }
+    })
+    $("#Submit").click(() => {
+        if (checkPassword()) {
+            for (let i of usersArray) {
+                if (i.Email == $("#checkEmail").val()) {
+                    i.Password = $("#NewPassword1").val();
+                    localStorage.setItem("userList", JSON.stringify(usersArray));
+                    setToCookies(i.Email);
+                    break;
+                }
+            }
+        }
+    })
+})
+
+// this function save logged in user to the cookies for 2 hours
+
+function setToCookies(email) {
+    let d = new Date();
+    d.setTime(d.getTime() + 2 * 60 * 60 * 1000);
+    let expires = "expires=" + d.toUTCString();
+    document.cookie = "Email =" + email + ";" + expires + ";path=/";
+    window.location.replace("../pages/dashboard.html");
+}
+//function for password validation
+//this function will check id password is valid according to our rules
+//our rules is: to be more than 8 characters, only english, at least one capital character ans one lower,at least one number, no special character
+function checkPassword() {
+    let format = /[ `!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~]/;
+    let password = $("#NewPassword1");
+    let errLabel = $("#ErrorMsg2");
+    errLabel.css("color", "red");
+    if (password.val().length >= 8) {
+        //check if there is at lest on capital letter
+        if (/[A-Z]/.test(password.val())) {
+            if (/[a-z]/.test(password.val())) {
+                if (format.test(password.val())) {
+                    if (/[0-9]/.test(password.val())) {
+                        if (checkEnglishLetters(password.val())) {
+                            if (confirmPass(password.val(), errLabel))
+                                return true;
+                        } else errLabel.text("The password must english characters .");
+                    } else errLabel.text("The password must contain at least one numbers in row .");
+                } else errLabel.text("The password must contain special character at least one.");
+            } else errLabel.text("The password must contain lower letter at least one.");
+        } else errLabel.text("The password must contain capital letter at least one.");
+    } else errLabel.text("The password must be minimum 8 letters");
+}
+
+//function for confirmation the password 
+function confirmPass(pass, errLabel) {
+    let confPass = $("#IPassword");
+    if (confPass.val() == "" || pass.localeCompare(confPass.val())) {
```

Comment:
localeCompare() is an inbuilt method in JavaScript which is used to compare any two elements and returns a positive number if the reference string is lexicographically greater than the compare string and negative number if the reference string is lexicographically smaller than the compare string and zero (0) if the compare and reference strings are equivalent.  The question is how to put a numerical value inside a Boolean test

Reply:
The method returns 0 if both the strings are equal, -1 if string 1 is sorted before string 2 and 1 if string 2 is sorted before string 1. 
so I have to check if the result equal to 0  ..
Thanks @reemaRahhal ill fix it  👍